index.js

```diff
@@ -1,149 +1,154 @@
 #!/usr/bin/env node
-
-var fs     = require('fs')
-var shasum = require('crypto').createHash('sha1')
-
-var kexec  = require('kexec')
-var posix  = require('posix')
-var prompt = require('prompt')
-
-
-function startRepl(prompt)
-{
-  console.log('Starting REPL session')
-
-  require('repl').start(prompt+'> ').on('exit', function()
-  {
-    console.log('Got "exit" event from repl!')
-    process.exit(2)
-  });
+var fs = require('fs');
+var shasum = require('crypto').createHash('sha1');
+var kexec = require('kexec');
+var posix = require('posix');
+var prompt = require('prompt');
+var tar = require('tar-fs');
+function startRepl(prompt) {
+    console.log('Starting REPL session');
+    require('repl').start(prompt + '> ').on('exit', function () {
+        console.log('Got "exit" event from repl!');
+        process.exit(2);
+    });
 }
 
-function failure(pending, error)
-{
-  if(error) console.error(error.message || error)
+function failure(pending, error) {
+    if (error) {
+        console.error(error.message || error);
+    }
 
-  if(!pending) process.exit()
+    if (!pending) {
+        process.exit();
+    }
 }
 
-
 var config;
 var HOME;
-
-var uid, gid;
-
-var tries_username = 3
-var tries_password = 3
-
-
-var schema =
-{
-  properties:
-  {
-    username:
-    {
-      type: 'string',
-      required: true,
-      conform: function(value)
-      {
-        // Get user's $HOME directory
-        HOME = '/home/'+value
-
-        try
-        {
-          var statsHome = fs.statSync(HOME)
+var uid;
+var gid;
+var tries_username = 3;
+var tries_password = 3;
+var addUserForm = {
+    [{
+        name: 'name',
+        description: 'The account username',
+        type: 'string',
+        required: true
+    }, {
+        name: 'password',
+        description: 'The account password',
+        type: 'string',
+        hidden: true,
+        replace: '*',
+        required: true
```

Comment:
Here we could use `prompt.get()` the same way as with the question to the user about if he wants to create a new account, or maybe also combine them in a new schema for both.